internal/elasticsearch/testdata/routing_rules/good/no_target_dataset_namespace_if/routing_rules.yml

```diff
@@ -0,0 +1,11 @@
+- source_dataset: missing_target_dataset
+  rules:
+    - if: ctx['aws.cloudwatch.log_stream'].contains('Test1')
+      namespace: default
```

Comment:
What would be the purpose of this reroute rule? Only to reroute to the data stream of another namespace? Would this make sense in a package?

Reply:
Yeah when `dataset` is not given, the original dataset part of the index name will be the fallback. For this case, it will just reroute to a different namespace. 
I'm only adding these cases for testing since `dataset` is not required by [reroute processor](https://www.elastic.co/guide/en/elasticsearch/reference/current/reroute-processor.html). Probably this is not a common use case. But for reroute processors to not have a given namespace, which will fallback to the old namespace, is common.

Reply:
Yes, I see the use case in a general sense of the reroute processor. But would a reroute rule without a target dataset be included in a package? Do you have an example for this?
I wonder if we should allow this in packages.

Reply:
Yeah for inside a package, I don't have any use case for without a target dataset. @felixbarny Do you know any example for this?

Reply:
It should even be allowed to create a completely empty reroute processor to route according to the dataset and namespace values in the document. For example, when the data_stream.* values have been modified via a `set` processor.

Reply:
Btw, @mrodm pointed me that these properties are mandatory now in the spec. If we want any of them to be actually optional we would need to change the spec first.
https://github.com/elastic/package-spec/blob/32eb741e298a29a505029e324b2e7c50e3031cef/spec/integration/data_stream/routing_rules.spec.yml#L52-L55

Reply:
> @felixbarny would such a rule be distributed in a package?

It might be, yes. I don't think it makes sense to restrict what's possible to do with packages. We might not need it... until we do need it. I think the package spec spec should go out of the way and not be too restrictive.

Reply:
I would prefer to have a use case for this before relaxing the check. In the meantime this is protecting developers from forgetting to set options that look needed in all the known cases.

Users can still use custom processing to add any routing rule they want.

Reply:
Thank you @jsoriano @felixbarny for your input!! For my use case, we definitely don't need to specify the `namespace` part. While we are thinking about use cases for optional `target_dataset`, should we first make the change in the spec and here to make `namespace` optional?

Reply:
Yep, namespace can make more sense as optional in rules distributed in packages. @kaiyan-sheng would you like to open the PR to change this in package spec?

Reply:
Yes I can do that :) Thanks!

Reply:
@jsoriano The PR for package spec is merged. Do we need to wait for the new spec version before moving forward on this PR?

Reply:
Well, CI in this PR is not failing, I guess we can go on, though it won't be possible to use this in packages till all pieces are released.